server/src/main/java/com/cloud/api/query/dao/TemplateJoinDaoImpl.java

```diff
@@ -290,15 +295,22 @@ public TemplateResponse newUpdateResponse(TemplateJoinVO result) {
 
     @Override
     public TemplateResponse setTemplateResponse(ResponseView view, TemplateResponse templateResponse, TemplateJoinVO template) {
+        Gson gson = new Gson();
 
         // update details map
-        if (template.getDetailName() != null) {
-            Map<String, String> details = templateResponse.getDetails();
-            if (details == null) {
-                details = new HashMap<>();
+        String key = template.getDetailName();
+        if (key != null) {
+            // FR37 TODO check properties and network prerequisites and if details is one of those fill those instead of detail
```

Comment:
addressed? do we still treat attributes other than properties in a special way and in fact, do we need to treat properties in a special way?